internal/pkg/describe/backend_service.go

```diff
@@ -10,55 +10,46 @@ import (
 	"sort"
 	"text/tabwriter"
 
-	"github.com/aws/copilot-cli/internal/pkg/config"
-	"github.com/aws/copilot-cli/internal/pkg/deploy"
 	"github.com/aws/copilot-cli/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
 	"github.com/aws/copilot-cli/internal/pkg/term/color"
 )
 
 // BackendServiceDescriber retrieves information about a backend service.
 type BackendServiceDescriber struct {
-	service         *config.Service
+	app             string
+	svc             string
 	enableResources bool
 
 	store                deployStoreSvc
-	svcDescriber         svcDescriber
+	svcDescriber         map[string]svcDescriber
 	initServiceDescriber func(string) error
 }
 
 // NewBackendServiceDescriber instantiates a backend service describer.
-func NewBackendServiceDescriber(app, svc string) (*BackendServiceDescriber, error) {
-	configStore, err := config.NewStore()
-	if err != nil {
-		return nil, fmt.Errorf("connect to store: %w", err)
-	}
-	meta, err := configStore.GetService(app, svc)
-	if err != nil {
-		return nil, err
-	}
-	deployStore, err := deploy.NewStore(configStore)
-	if err != nil {
-		return nil, fmt.Errorf("connect to deploy store: %w", err)
-	}
-	opts := &BackendServiceDescriber{
-		service:         meta,
+func NewBackendServiceDescriber(opt *NewServiceDescriberOption) (*BackendServiceDescriber, error) {
+	describer := &BackendServiceDescriber{
+		app:             opt.App,
+		svc:             opt.Svc,
 		enableResources: false,
-		store:           deployStore,
+		store:           opt.DeployStore,
+		svcDescriber:    make(map[string]svcDescriber),
 	}
-	opts.initServiceDescriber = func(env string) error {
-		d, err := NewServiceDescriber(app, env, svc)
+	describer.initServiceDescriber = func(env string) error {
+		opt.Env = env
```

Comment:
it's because previously we have a bug. See it [here](https://github.com/aws/copilot-cli/pull/1103/files#diff-88fdaab0450f79a9400f9ba9c6d7c280R121). When we have multiple environments and specify enableResource flag then the describer will always be with the last environment's manager role. it is basically used to cache describer in different account so that we don't need to init them again when describing resources.